src/UserSpaceInstrumentation/RegisterState.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "UserSpaceInstrumentation/RegisterState.h"
+
+#include <cpuid.h>
+#include <elf.h>
+#include <errno.h>
+#include <sys/ptrace.h>
+#include <sys/uio.h>
+
+#include <string>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/SafeStrerror.h"
+#include "absl/strings/str_format.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace {
+
+// Return the size of the XSave area on this cpu.
+[[nodiscard]] ErrorMessageOr<size_t> GetXSaveAreaSize() {
+  uint32_t eax = 0;
+  uint32_t ebx = 0;
+  uint32_t ecx = 0;
+  uint32_t edx = 0;
+  if (!__get_cpuid(1, &eax, &ebx, &ecx, &edx) || !(ecx & bit_XSAVE)) {
+    return ErrorMessage("XSAVE is not supported by Cpu.");
+  }
+  if (!__get_cpuid_count(0x0d, 0x00, &eax, &ebx, &ecx, &edx)) {
+    return ErrorMessage("XSAVE is not supported by Cpu.");
```

Comment:
This error message is the same as the one above, but in `GetAvxOffset` the error message for `__get_cpuid_count` is different. Is this on purpose?

Reply:
Yes, the cpuid leaf 0x01 returns in ecx & bit_XSAVE wheather the processor supports XSAVE. The subleave 0x0d, 0x00 return the maxímum size of the xsave area in ecx. 
Both can only fail when the processor is acient and does not support xsave. So the messages are fine I think.

I'm unsure if the second one can happen; probably the first will already catch all cases where the second call fails.

For the Avx case there might be different reasons (well, theoretically, I'm not sure if the second one can be hit).